collections/_article/introducing-reverse-z.md

```diff
@@ -0,0 +1,135 @@
+
+We are certain that the vast majority of users will not run into any compatibility breakage. For most people this change is all good with no downside. 
+
+You may need to tweak your shaders if you use a custom spatial shader that:
+
+- Writes to ``POSITION`` in the vertex processor function
+
+- Writes to ``DEPTH``  in the fragment processor function
+
+- Reads from depth_texture
+
+- Operates in clip space
+
+In most cases when working with ``POSITION``, ``DEPTH``, or the depth_texture, you won't need to make any shader changes as long as you are transforming the values into/out of clip space using the ``PROJECTION_MATRIX`` or the ``INV_PROJECTION_MATRIX``. In those cases, the transformation is handled for you and your shader will continue to work. 
+
+Let's look at the cases one-by-one to see what sort of shaders will break and how to fix them.
+
+
+### Writes to ``POSITION``
+
+When you write to ``POSITION`` in Godot, you are bypassing the built-in vertex transformation and writing a clip space position directly. This can be helpful to optimize a vertex shader, or to achieve certain effects (like having a mesh stay fixed in the camera view). Commonly users wrote the following line:
+
+```
+POSITION = vec4(VERTEX, 1.0);
+```
+
+This line is even reproduced in our [documentation](https://docs.godotengine.org/en/4.2/tutorials/shaders/advanced_postprocessing.html). It relies on users supplying a QuadMesh with a width and height of 2 to fill the entire screen. The code assumes that:
+
+1. All vertices of the mesh have a Z-coordinate of 0 (which is true for the QuadMesh), and
+
+2. A clip space value of 0 corresponds to the near plane.
+
+We have broken assumption 2 by flipping the definition of the near plane and the far plane, so this code no longer works. Instead, users need to write:
+
+```
+POSITION = vec4(VERTEX.xy, 1.0, 1.0);
+```
+
+And that's it!
+
+Since this code is used so widely, we are adding a special [warning](https://github.com/godotengine/godot/pull/90587) for it so the engine can give you a heads up if your code will potentially break in 4.3.
+
+Importantly, writes to ``POSITION`` will only break if they are writing values in directly in clip space. If you are transforming values from view space using the ``PROJECTION_MATRIX``, no code changes are necessary. For example, the following code will continue to work:
+
+```
+POSITION = PROJECTION_MATRIX * MODELVIEW_MATRIX * vec4(VERTEX, 1.0);
+```
+
+### Writes to ``DEPTH``
+
+Writing to ``DEPTH`` comes with the same warnings as writing to ``POSITION``. If the clip space value comes from transforming a view space value with the ``PROJECTION_MATRIX``, then no changes are necessary. 
+
+```
+// This will continue to work
+vec4 clip_pos = PROJECTION_MATRIX * vec4(VERTEX, 1.0);
+clip_space.xyz /= clip_space.w;
+DEPTH = clip_space.z;
+
+// This will need to change
+DEPTH = 0.0; // Needs to change to 1.0
```

Comment:
Follow Godot comment style guide:

```suggestion
// This will continue to work.
vec4 clip_pos = PROJECTION_MATRIX * vec4(VERTEX, 1.0);
clip_space.xyz /= clip_space.w;
DEPTH = clip_space.z;

// This will need to change.
DEPTH = 0.0;  // Needs to change to 1.0.
```